server/services/ThreatIntelService.ts

```diff
@@ -0,0 +1,342 @@
+        params
+      );
+      const sources = searchSourcesResponse.hits.hits.map((hit: ThreatIntelSourceSearchHit) => ({
+        id: hit._id,
+        ...hit._source.source_config,
+      }));
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: true,
+          response: sources,
+        },
+      });
+    } catch (error: any) {
+      console.error('Security Analytics - ThreatIntelService - searchThreatIntelSources:', error);
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: false,
+          error: error.message,
+        },
+      });
+    }
+  };
+
+  deleteThreatIntelSource = async (
+    context: RequestHandlerContext,
+    request: OpenSearchDashboardsRequest<{ sourceId: string }, any>,
+    response: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<ServerResponse<any> | ResponseError>> => {
+    try {
+      const params = { sourceId: request.params.sourceId };
+      const client = this.getClient(request, context);
+      const deleteRes: any = await client(
+        CLIENT_THREAT_INTEL_METHODS.DELETE_THREAT_INTEL_SOURCE,
+        params
+      );
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: true,
+          response: deleteRes,
+        },
+      });
+    } catch (error: any) {
+      console.error('Security Analytics - ThreatIntelService - deleteThreatIntelSource:', error);
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: false,
+          error: error.message,
+        },
+      });
+    }
+  };
+
+  refreshThreatIntelSource = async (
+    context: RequestHandlerContext,
+    request: OpenSearchDashboardsRequest<{ sourceId: string }, any>,
+    response: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<ServerResponse<any> | ResponseError>> => {
+    try {
+      const params = { sourceId: request.params.sourceId };
+      const client = this.getClient(request, context);
+      const refreshRes: any = await client(
+        CLIENT_THREAT_INTEL_METHODS.REFRESH_THREAT_INTEL_SOURCE,
+        params
+      );
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: true,
+          response: refreshRes,
+        },
+      });
+    } catch (error: any) {
+      console.error('Security Analytics - ThreatIntelService - refreshThreatIntelSource:', error);
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: false,
+          error: error.message,
+        },
+      });
+    }
+  };
+
+  getThreatIntelIocs = async (
+    context: RequestHandlerContext,
+    request: OpenSearchDashboardsRequest<{}, any>,
+    response: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<ServerResponse<any> | ResponseError>> => {
+    try {
+      const params = { feed_id: request.query.feedIds };
```

Comment:
feedIds? should be a ist?

Reply:
not a list for the same reason that we fixed a bug in correlations